tests/unit/test_project.py

```diff
@@ -517,7 +520,7 @@ def test_project_name_valid(yaml_loaded_data, valid_name):
 
 
 @pytest.mark.parametrize(
-    "invalid_name", ["AAA", "0aaa", "a", "a--a", "aa-", "a:a", "a/a", "a@a", "a_a"]
+    "invalid_name", ["", "AAA", "a--a", "aa-", "a:a", "a/a", "a@a", "a_a"]
```

Comment:
If the answer to my question about about the 40-character name limit is okay, do we want to test it here? (Maybe not if it's incidental and we're happy for that requirement to change upstream and simply adopt it.)

Likewise, should we add the 40-char limit to [the reference docs](https://canonical-ubuntu-documentation-library--957.com.readthedocs.build/rockcraft/957/reference/rockcraft.yaml/#name)?